### DISCUSSION TITLE
`fill` channel not working

### DISCUSSION THREAD

**Discussion by winner106** [observablehq/plot#1720] [Q&A] [answered]
[https://observablehq.com/d/1541115f66f2a3eb](https://observablehq.com/d/1541115f66f2a3eb)

![image](https://github.com/observablehq/plot/assets/50519642/f2ccbfec-6af3-4ede-9b86-394a973492a6)

![image](https://github.com/observablehq/plot/assets/50519642/8044add5-4416-4019-b4af-ea6a78d86fe0)

Thank you very much.

**Comment by mbostock** [comment]
All color channels share the same color scale. So if you want to pass the startingChar column as the fill channel, you should make sure no other color channels are passing inconsistent data, such as `fill: "age"` in the code you posted. Try disabling those other channels first and then add your new fill channel.

**Reply by winner106** [reply]
Thank you for your guidance. I have now resolved the issue by utilizing a custom color scale. The chart is great, but I am facing challenges with overlapping data labels. I look forward to the possibility of assigning values to the dx and dy channels in the future, as it should facilitate easier resolution of this matter.

![image](https://github.com/observablehq/plot/assets/50519642/6fe9ee61-27d8-443a-98b8-0e0c3fda832f)

**Reply by mbostock** [reply, marked as answer]
We have no plans to allow **dx** and **dy** to be specified as channels. These are constant options. If you to specify different values of **dx** and **dy**, you can create multiple marks, optionally using a [filter](https://observablehq.com/plot/transforms/filter) to control which data gets which values of **dx** and **dy**.

Edit: As I described earlier in https://github.com/observablehq/plot/issues/1528#issuecomment-1537657203.
